ts/features/bonus/cdc/screens/CdcBonusRequestSelectResidence.tsx

```diff
@@ -98,12 +101,22 @@ const CdcBonusRequestSelectResidence = () => {
           })}
           rightButton={confirmButtonProps(
             () => {
+              dispatch(
+                cdcSelectedBonusAction(
+                  cdcSelectedBonus?.map(b => ({
+                    year: b.year,
+                    residence: isResidentInItaly[b.year]
+                  }))
+                )
+              );
               navigation.navigate(CDC_ROUTES.BONUS_REQUESTED);
             },
             I18n.t("global.buttons.continue"),
             undefined,
             undefined,
-            !cdcSelectedBonus.every(b => isResidentInItaly[b.year] === "italy")
+            !cdcSelectedBonus.every(b =>
+              Object.keys(isResidentInItaly).includes(b.year)
```

Comment:
This could be simplified in complexity doing:

```ts
!cdcSelectedBonus.every(b => !!isResidentInItaly[b.year])
```

and _maybe_ it could be more readable like:

```ts
cdcSelectedBonus.some(b => !isResidentInItaly[b.year])
```

Do you think it's worth it, or I'm missing something logically? 🙂

Reply:
Sounds good to me, thank you! Refactored in [c365840](https://github.com/pagopa/io-app/pull/3955/commits/c36584012182d9d829354ef2d044bbece6d9b3f5)